drivers/storage/portworx/component/telemetry.go

```diff
@@ -59,16 +94,496 @@ func (t *telemetry) Delete(cluster *corev1.StorageCluster) error {
 	if err := k8sutil.DeleteConfigMap(t.k8sClient, TelemetryConfigMapName, cluster.Namespace, *ownerRef); err != nil {
 		return err
 	}
+
+	if err := t.deleteMetricsCollector(cluster.Namespace, ownerRef); err != nil {
+		return err
+	}
+
+	t.MarkDeleted()
 	return nil
 }
 
-func (t *telemetry) MarkDeleted() {}
+func (t *telemetry) MarkDeleted() {
+	t.isCollectorDeploymentCreated = false
+}
 
 // RegisterTelemetryComponent registers the telemetry  component
 func RegisterTelemetryComponent() {
 	Register(TelemetryComponentName, &telemetry{})
 }
 
+func (t *telemetry) deleteMetricsCollector(namespace string, ownerRef *metav1.OwnerReference) error {
+	if err := k8sutil.DeleteServiceAccount(t.k8sClient, CollectorServiceAccountName, namespace, *ownerRef); err != nil {
+		return err
+	}
+
+	if err := k8sutil.DeleteConfigMap(t.k8sClient, CollectorProxyConfigMapName, namespace, *ownerRef); err != nil {
+		return err
+	}
+
+	if err := k8sutil.DeleteConfigMap(t.k8sClient, CollectorConfigMapName, namespace, *ownerRef); err != nil {
+		return err
+	}
+
+	if err := k8sutil.DeleteRole(t.k8sClient, CollectorRoleName, namespace, *ownerRef); err != nil {
+		return err
+	}
+
+	if err := k8sutil.DeleteRoleBinding(t.k8sClient, CollectorRoleBindingName, namespace, *ownerRef); err != nil {
+		return err
+	}
+
+	if err := k8sutil.DeleteDeployment(t.k8sClient, CollectorDeploymentName, namespace, *ownerRef); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *telemetry) deployMetricsCollector(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	if err := t.createServiceAccount(cluster.Namespace, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createProxyConfigMap(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorConfigMap(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorRole(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorRoleBinding(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	if err := t.createCollectorDeployment(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *telemetry) createCollectorDeployment(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	deployment, err := t.getCollectorDeployment(cluster, ownerRef)
+	if err != nil {
+		return err
+	}
+
+	existingDeployment := &appsv1.Deployment{}
+	err = t.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      CollectorDeploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	modified := !util.DeploymentDeepEqual(deployment, existingDeployment)
+
+	if !t.isCollectorDeploymentCreated || modified {
+		logrus.WithFields(logrus.Fields{
+			"isCreated": t.isCollectorDeploymentCreated,
+			"modified":  modified,
+		}).Info("will create/update the deployment.")
```

Comment:
nit: Let's keep this as trace log. It will create too much unnecessary log. Unfortunately our default log is debug right now, so can't keep this as debug.

Reply:
This wont print too many, as it is only when it is created or modified. not in every reconcile loop